flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/RemoteStreamEnvironment.java

```diff
@@ -213,20 +210,6 @@ private static Configuration getEffectiveConfiguration(
 		return effectiveConfiguration;
 	}
 
-	@Override
-	public JobExecutionResult execute(StreamGraph streamGraph) throws Exception {
-		try {
-			return super.execute(streamGraph);
-		}
-		catch (ProgramInvocationException e) {
-			throw e;
-		}
-		catch (Exception e) {
-			String term = e.getMessage() == null ? "." : (": " + e.getMessage());
-			throw new ProgramInvocationException("The program execution failed" + term, e);
```

Comment:
Let's say that FLINK-15090 requires remove usage of `ProgramInvocationException` which belongs to `flink-client` which will then be no longer a dependency of `flink-streaming`. I don't think user should rely on `ProgramInvocationException` which itself doesn't keep consistency(always thrown) among environments. It would be just for display and no harm we use the `super` implementation.

Reply:
I think this change is not as simple as it might look like. By changing the thrown exception we are effectively changing public API here. Such a change should definitely not happen as part of a hotfix commit!

Admittedly, the signature says that we are throwing an `Exception` instead of a `ProgramInvocationException` but until now the `RemoteStreamEnvironment` contract that it only throws `ProgramInvocationExceptions`.